contrib/squash_commit.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash -eEu
+#
+# Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+# See file LICENSE for terms.
+#
+# Prepare one commit after PR has been approved.
+#
+
+message=
+parent=
+remote=
+
+usage() {
+    echo "usage: $0 -m <commit_message> [-r <remote_branch>] [-p <new_parent>]" >&2
+    echo "" >&2
+    echo "Arguments" >&2
+    echo "  -m <commit_message>     commit message to use" >&2
+    echo "  -r <remote_branch>      upstream master branch to merge into" >&2
+    echo "  -p <new_parent>         new base to use for squashed commit" >&2
+    echo "" >&2
+    echo "Sample" >&2
+    echo "    $0 -m \"Commit message\" -r upstream/master" >&2
+    echo "    $0 -m \"Commit message\" -p adef30" >&2
+    exit 1
+}
+
+param_checks() {
+    if [ -z "$message" ]
+    then
+        usage
+    fi
+
+    if [ -z "$parent" ] && [ -z "$remote" ]
+    then
+        echo "error: parent must be specified when remote is missing (-p)" >&2
+        exit 1
+    fi
+
+    if [ -z "$parent" ]
+    then
+        parent=$(git merge-base HEAD "$remote")
+    fi
+
+    typ=$(git cat-file -t "$parent" || :)
+
+    if [ "$typ" != "commit" ]
+    then
+        echo "error: parent sha1 $parent is not a commit" >&2
+        exit 1
+    fi
+}
+
+while getopts "hm:r:p:" o; do
+    case "$o" in
+    m)
+        message="$OPTARG"
+        ;;
+    r)
+        remote="$OPTARG"
+        ;;
+    p)
+        parent="$OPTARG"
+        ;;
+    *)
+        usage
+        ;;
+    esac
+done
+
+if [ ! -f .git ] && [ ! -d .git ]
+then
+    echo "error: run the script from top directory" >&2
+    exit 1
+fi
+
+param_checks
+
+git update-index --really-refresh
+if ! git diff-index --quiet HEAD
+then
+    echo "error: working directory has changes (run 'git reset --hard'?)" >&2
+    exit 1
+fi
+
+revision=$(git rev-parse HEAD)
+echo "Using revision $revision"
+
+set -x
+git clean -fdx
```

Comment:
imo, this is not really required and may delete some oimportant info (logs, diffs, core, etc)

Reply:
for commits with added files, we later do add --all, so we need to clean the tree first. but agree on the deletion, pushing simpler approach to avoid the need to clean files.